lib/openstudio-standards/standards/Standards.Model.rb

```diff
@@ -3054,11 +3054,24 @@ def model_add_material(model, material_name)
   # @return [OpenStudio::Model::Construction] construction object
   # @todo make return an OptionalConstruction
   def model_add_construction(model, construction_name, construction_props = nil, surface = nil)
+    intended_surface_type = construction_props&.[]('intended_surface_type') || ''
+
     # First check model and return construction if it already exists
     model.getConstructions.sort.each do |construction|
       if construction.name.get.to_s == construction_name
         OpenStudio.logFree(OpenStudio::Debug, 'openstudio.standards.Model', "Already added construction: #{construction_name}")
-        return construction
+        valid = true
+        if !surface.nil?
+          if intended_surface_type == 'GroundContactFloor' && construction.iddObjectType.valueName != 'OS_Construction_FfactorGroundFloor'
+            valid = false
+          elsif intended_surface_type == 'GroundContactWall' && construction.iddObjectType.valueName != 'OS_Construction_CfactorUndergroundWall'
+            valid = false
+          end
+        end
+        if valid
+          return construction
+        end
+        OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.Model', "Already added construction: '#{construction_name}' but its type '#{construction.iddObjectType.valueName}' is not valid for the intended surface type '#{intended_surface_type}'. A new construction will be created.")
```

Comment:
First error handling:

If you can find the construction by name, BUT it's in the incorrect type, then proceed to create it anyways.